ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java

```diff
@@ -434,6 +458,17 @@ protected void writeContent(AsyncWritableChannel writeChannel, ReadIntoCallbackW
   }
 
   /**
+   * Switches auto reading from the channel on and off.
+   * @param autoRead {@code true} if auto read has to be switched on. {@code false} otherwise.
+   */
+  protected void setAutoRead(boolean autoRead) {
+    channel.config().setAutoRead(autoRead);
+    channel.config().setMaxMessagesPerRead(autoRead ? savedMaxMessagesPerRead : 1);
```

Comment:
I can make it the other way round but from what I can gather from the documentation, if autoRead is set to `true`, the channel is read continuously and maxMessagesPerRead should not matter. It only matters when autoRead is `false` and you care about how much you read once you call `channel.read()`.